### DISCUSSION TITLE
Initial Launch URL

### DISCUSSION THREAD

**Discussion by GoodluckH** [ShafSpecs/remix-pwa#130] [Q&A] [answered]
Following up on the Discord discussion here: https://discord.com/channels/770287896669978684/1136740580626542683/1164272621177819168

I was wondering if there's a way to set the default url of a newly installed pwa?

So let's say i click "Install" on "/app", this will prompt my browse to close the existing window and open the "/app" page on installed pwa.

One PWA I found that has achieved this is [llamalife.co](https://llamalife.co/). If you are on their home page and install the PWA, once you click on "Install" button, the page suddenly redirect to "/howToInstall" and then close to launch the PWA.

Not sure how to implement this behavior.

**Comment by ShafSpecs** [comment, marked as answer]
Typically, you would want to use the `appinstalled` event available on the `Window` interface. In your root, or layout route, you can do something like this:
```ts
useEffect(() ={
 if (typeof window === 'undefined') return;

 window.addEventListener('appinstalled', () => {
  navigate('/')
 })

 return () => /* remove the event listener here */
}, [])
```
Something like this should get the job done :+1:. Lemme know if it worked for you or you found a better approach. Have a look at: https://developer.mozilla.org/en-US/docs/Web/API/Window/appinstalled_event whilst at it.

**Reply by GoodluckH** [reply]
This worked very well! Thank you!
